python/cuml/dask/datasets/regression.py

```diff
@@ -40,9 +45,173 @@ def create_rs_generator(random_state):
+    chunks = [client.submit(_f_order_standard_normal, chunksize, ncols, dtype,
+                            chunk_seeds[idx], workers=[chunks_workers[idx]],
+                            pure=False)
+              for idx, chunksize in enumerate(chunksizes)]
+
+    chunks_dela = [da.from_delayed(dask.delayed(chunk),
+                                   shape=(chunksizes[idx], ncols),
+                                   meta=cp.zeros((1)), dtype=dtype)
+                   for idx, chunk in enumerate(chunks)]
+    return da.concatenate(chunks_dela, axis=0)
+
+
+def _data_from_multivariate_normal(seed, covar, n_samples, n_features, dtype):
+    mean = cp.zeros(n_features)
+    local_rs = cp.random.RandomState()
+    return local_rs.multivariate_normal(mean, covar, n_samples,
+                                        dtype=dtype)
+
+
+def data_from_multivariate_normal(client, rs, covar, chunksizes, n_features,
+                                  dtype):
+    workers = list(client.has_what().keys())
+
+    n_chunks = len(chunksizes)
+    chunks_workers = (workers * n_chunks)[:n_chunks]
+
+    chunk_seeds = rs.permutation(len(chunksizes))
+
+    data_parts = [client.submit(_data_from_multivariate_normal,
+                                chunk_seeds[idx], covar,
+                                chunksizes[idx], n_features,
+                                dtype, workers=[chunks_workers[idx]],
+                                pure=False)
+                  for idx, chunk in enumerate(chunksizes)]
+
+    data_dela = [da.from_delayed(dask.delayed(chunk),
+                                 shape=(chunksizes[idx], n_features),
+                                 meta=cp.zeros((1)), dtype=dtype)
+                 for idx, chunk in enumerate(data_parts)]
+    return da.concatenate(data_dela, axis=0)
+
+
+def _f_order_shuffle(X, y, n_samples, seed, features_indices):
+    local_rs = cp.random.RandomState(seed=seed)
+    samples_indices = local_rs.permutation(n_samples)
+
+    X[...] = X[samples_indices, :]
+    X[...] = X[:, features_indices]
+
+    y[...] = y[samples_indices, :]
+    return X, y
+
+
+def f_order_shuffle(client, rs, X, y, chunksizes, n_features,
+                    features_indices, n_targets, dtype):
+    X_parts = client.sync(_extract_partitions, X)
+    y_parts = client.sync(_extract_partitions, y)
+
+    chunk_seeds = rs.permutation(len(chunksizes))
+
+    shuffled = [client.submit(_f_order_shuffle, X_part, y_parts[idx][1],
+                              chunksizes[idx],
+                              chunk_seeds[idx], features_indices,
+                              workers=[w], pure=False)
+                for idx, (w, X_part) in enumerate(X_parts)]
+
+    X_shuffled = [client.submit(get_X, f, pure=False)
+                  for idx, f in enumerate(shuffled)]
+    y_shuffled = [client.submit(get_labels, f, pure=False)
+                  for idx, f in enumerate(shuffled)]
+
+    X_dela = [da.from_delayed(dask.delayed(Xs),
```

Comment:
The `X_dela` and `y_dela` look like like they duplicate a lot of the same calls. Can we make a little private function for this for this such as `_dask_array_from_delayed(X_shuffled, chunksizes, n_features, dtype)`?